### PR TITLE
Search updates for top navigation

### LINKED ISSUE #4376: [New navigation] Search (on mobile)

### PR DESCRIPTION
**Pull request by bartaz**
## Done

Update search for new nav (mobile search dropdown)

Fixes #4376 

## QA

- Open [demo](https://vanilla-framework-4396.demos.haus/docs/examples/patterns/navigation/search-dark)
- Make sure the expanding search works as expected
  - check if search can be toggled on/off
  - check if it works as expected on both mobile and desktop
  - check light and dark themes
    - https://vanilla-framework-4396.demos.haus/docs/examples/patterns/navigation/search-light
    - https://vanilla-framework-4396.demos.haus/docs/examples/patterns/navigation/search-dark
  - check deprecated example to make sure it still works fine
    - https://vanilla-framework-4396.demos.haus/docs/examples/patterns/navigation/deprecated
- Review updated documentation:
  - https://vanilla-framework-4396.demos.haus/docs/patterns/navigation#expanding-search-box

## Screenshots

new-nav

### CODE REVIEW

File: templates/docs/examples/patterns/navigation/search-dark-new.html [hunk 1]
```diff
@@ -9,9 +9,12 @@
 <header id="navigation" class="p-navigation is-dark">
   <div class="p-navigation__row">
     <div class="p-navigation__banner">
-      <div class="p-navigation__logo">
-        <a class="p-navigation__item" href="#">
-          <img class="p-navigation__image" src="https://assets.ubuntu.com/v1/3c7954dd-logo-canonical-white.svg" alt="" width="95" />
+      <div class="p-navigation__tagged-logo">
+        <a class="p-navigation__link" href="#">
+          <div class="p-navigation__logo-tag">
+            <img class="p-navigation__logo-icon" src="https://assets.ubuntu.com/v1/82818827-CoF_white.svg" alt="">
```
Comment (fitojb) [inline templates/docs/examples/patterns/navigation/search-dark-new.html]: The Circle of Friends is going to be a symbol for the Canonical company too?!

Comment (bartaz) [reply templates/docs/examples/patterns/navigation/search-dark-new.html]: It is a placeholder for now. But as far as I understand some smaller websites/products that don't have their own logo may use this instead when new branding is rolled out.

**Comment by lyubomir-popov** [comment]
Looks great @bartaz. Scope creep: can we make the icon match the text colour? 
image

**Comment by bethcollins92** [comment]
The new examples seem to be working fine but the deprecated expanding search isn't working on the demo: 

Kapture 2022-04-12 at 09 41 49

Also, I'm not sure the best way to fix this or if it's important, but the suggested search items box looks disconnected from the search input

Screenshot 2022-04-12 at 09 37 40

Screenshot 2022-04-12 at 09 59 15

**Comment by bartaz** [comment]
Good catch with deprecated search, but I'm not sure what happens with autocomplete, seems to be fine for me:
image

**Review by bethcollins92 — COMMENTED** [review]
The deprecated demo LGTM now, not sure what's happening with the autocomplete on mine, maybe @albertkol or @andesol could take a look and check it looks good to them?

**Pull request merged**